test/it/client-listUsers.js

```diff
@@ -51,7 +67,7 @@ describe('client.listUsers().each()', () => {
   it('should return User models', () => {
     return client.listUsers().each(user => {
```

Comment:
Yes, I have tested this with taped responses. So, the tests do make calls that would be handled by test-server.
Regarding your second question, the test isn't really making repeated calls to `/api/v1/users`. It just makes 1 call, which returns a collection and then we iterate over it to ensure we have the right models. So, your implementation doesn't really matter as long as you make 1 call to `/api/v1/users`. Your test can then work on the collection return in any manner you deem right.